fedbiomed/researcher/federated_workflows/jobs/_training_job.py

```diff
@@ -46,7 +46,8 @@ def __init__(self,
         Args:
             nodes: A dict of node_id containing the nodes used for training
             keep_files_dir: Directory for storing files created by the job that we want to keep beyond the execution
-                of the job. Defaults to None, files are not kept after the end of the job.
+                of the job. Defaults to None, files are not kept after the end of the job. Usually, files are kept in
```

Comment:
We are changing a bit the semantics for `keep_files_dir` by removing the default value. 

From the docstring (although I don't know if it was actually true in the code) it seems that `keep_files_dir=None` had the meaning of  "do not keep files at the end of Job". However, I'm a bit confused if this even makes sense or not. 

@mvesin are you the most up-to-date person on this? Do you remember if `keep_files_dir=None` is actually implemented and used somewhere? Is it ok if we remove this default value, and we force it to be a `str`?

Reply:
It is what we have discussed and decided  with mvesin: `keep_files_dir` should be used each time a `job` is created and then used. there is no place `keep_files_dir=None` is actually used: each time we are passing a path and are saving things within it,

Reply:
Thank you @sharkovsky 
I confirm this modification is intentional (and discussed with  @ybouilla )

The "do not keep the file at the end" is an obsolete option we did not dare to remove before this refactoring.